### DISCUSSION TITLE
best way to set tls certificates directly in memory?

### DISCUSSION THREAD

**Discussion by gabor** [jackc/pgx#1903] [Q&A]
when using the tls certificate options in the connection string, it's assumed the certificates are on disk. in my case i have them in memory (i basically have a `*tls.Config` already).
what is the best way to use such certificates?

i found a way that works:
```go
connstr := "connection string without tls things"
conf := pgx.ParseConfig(connstr)
conf.TLSConfig = tlsConf
db := pgxstdlib.OpenDB(conf)
```

but i also saw this warning:  https://github.com/jackc/pgx/blob/5c63f646f820ca9696fc3515c1caf2a557d562e5/pgconn/config.go#L204-L209

"Ensure there are no stale fallbacks when manually setting TLSConfig."

is there maybe a more robust way to do this?

if not, how should i "ensure there are no stale fallbacks"? before replacing `conf.TLSConfig`, should i check if it's `nil` or not, and return an error if it's not `nil`?

**Comment by jackc** [comment]
> ensure there are no stale fallbacks

Make sure that `Config.Fallbacks` is empty.
